### ISSUE TITLE
grldr 20230922 & 20231014: can not 'root (cd)' anymore after booting from iso

### ISSUE THREAD

**Issue by deomsh** [chenall/grub4dos#431]
Problem with newest grldr versions 20230922 & 20231014: can not 'root (cd)' anymore after booting from iso (El Torito boot). Up to version 20230610 NO problems (tested grldr versions without the problem: 20220915, 20221026, 20221128, 20222122, 20230306, 20230329, 20230610). 

If I change the root after booting an iso with grldr, I can't use 'root (cd)', ALWAYS Error 23. Only 'root (0xE0)' is working (if 0xE0 is the drive number). Afterwards '%@root%' is still '(cd)'.

Same problem in Limbo x86 (QEMU-based).

See the print-screens:

![VBox After booting from iso grub4dos 20231014, problem with root (cd), NOT with root (0xE0) with colored marks](https://github.com/chenall/grub4dos/assets/67714723/367591be-491d-4993-9558-0f019ceb6fd5)
![VBox After booting from iso grub4dos 20230922, problem with root (cd), NOT with root (0xE0) with colored marks](https://github.com/chenall/grub4dos/assets/67714723/e2f0f35d-cd27-4425-bdd7-275f250aa7e2)
![VBox After booting from iso with grub4dos version 20230610, NO problem with root (cd), or drvnum (0xE0) with colored marks](https://github.com/chenall/grub4dos/assets/67714723/0d6c9137-a6d1-4963-bae4-8328cdfd122b)

BTW: no problems with grub4efi (only using '(cd0)' instead of '(cd)' and other drive number):

![VBox After booting from iso with grub4efi 20231016, NO problem with root (cd0), or drvnum (0xA0) with colored marks](https://github.com/chenall/grub4dos/assets/67714723/1c5eaab4-f0bb-43c8-90b9-31ecb4a76ecf)

**Comment by yaya2007** [comment]
这个问题已经解决。

**Comment by deomsh** [comment]
Thanks, I hope there is a new release soon

**Comment by yaya2007** [comment]
The 2023-11-27 version has been corrected.

**Comment by deomsh** [comment]
I understand. But a new grub4dos release would be nice. For now there are NO grub4dos releases after 2021, only grub4efi....

**Comment by deomsh** [comment]
Thanks to post #433 I found the download for Grub4dos 2023-11-27. 

I tested for %@root% after booting to CD. Device (cd) is gone, now always (0xe0). In scripts device=%~d1 etcetera included. So thanks a lot!

Side-note: grub4dos 2023 versions seems to need more memory. In Virtual Box 7 I needed at least 131MB, otherwise grub4dos scripts gave ERROR 13, grubutil 'fat' needed at least 259MB (below ERROR 13 too).
With memory below 131MB the message 'A20 debug: C806 done! ...' was long visible, while starting grub4dos.
